cmds/dhclient/dhclient.go

```diff
@@ -158,15 +160,25 @@ func dhclient4(iface netlink.Link, numRenewals int, timeout time.Duration, retry
 				routerName := net.IP(gwData).String()
 				debug("routerName %v", routerName)
 				r := &netlink.Route{
-					Dst:       &net.IPNet{IP: packet.GIAddr()},
 					LinkIndex: iface.Attrs().Index,
 					Gw:        net.IP(gwData),
 				}
 
-				if err := netlink.RouteReplace(r); err != nil {
+				if err := netlink.RouteAdd(r); err != nil {
```

Comment:
How does this work with renewal requests? Will we just keep adding routes? I don't know which take priority and how.

Also, we should probably do this for v6 as well?

Reply:
I'm not ready to do this for IP6 until we have a reliable replacement. As for renewals, all our IP4 clients seem to get perpetual leases so far. I don't have a good test for that.

As it happens, due to other bugs in this code, the error seems to have always been occurring and never being logged. It's still a mess, just a much better mess. I'm convinced it's still not completely right.

Reply:
But finding out what's wrong is not easy.

Reply:
"I'm not ready to do this for IP6 until" we have a reliable server/client setup. Sorry, that was really confusing wording.

Reply:
is it possible there's a problem in the netlink code? Tbh I don't know how this netlink socket route adding stuff works.

Reply:
who knows. The netlink interface in linux is so incredibly awful that very few people really know how to make it work. That's why everyone always spawns programs for this stuff instead of just writing to it. It's like everyone forked off dd to do file I/O.

Reply:
one more time ... sorry

Reply:
do we get perpetual leases? I thought our testing dhcp server has a 5m lease.

Reply:
don't know but one thing I did observe -- IIRC this code was never working right.